racksdb/web/ui/src/views/DatacenterRoomView.vue

```diff
@@ -55,26 +84,34 @@ function listInfrastructures(rackName: string) {
 }
 
 async function getDatacenters() {
-  datacenters.value = await racksDBAPI.datacenters()
-  datacenterDetails.value = datacenters.value.filter(
+  // Get datacenters and filter using the prop datacenterName
+  let datacenterDetails = (await racksDBAPI.datacenters()).filter(
     (datacenter) => datacenter.name === props.datacenterName
   )[0]
-}
 
-async function getInfrastructures() {
-  infrastructures.value = await racksDBAPI.infrastructures()
+  // Loop on datacenterDetails to get all the rack it contains
+  datacenterDetails.rooms.forEach((room) => {
+    room.rows.forEach((row) => {
+      row.racks.forEach((rack) => {
+        rackDetails.value.push({
+          name: rack.name,
+          fillrate: rack.fillrate,
+          room: room.name
+        })
+      })
+    })
+  })
 }
 
-async function getRacks() {
-  racks.value = await racksDBAPI.racks()
-  rackDetails.value = racks.value.filter((rack) => rack.room === props.datacenterRoom)
+async function getInfrastructures() {
+  infrastructures = await racksDBAPI.infrastructures()
 }
 
 onMounted(() => {
   getDatacenters()
   getInfrastructures()
-  getRacks()
   getInfrastructureImg()
+  displayRacks()
```

Comment:
I think it is useless to call `displayRacks()` here, VueJS will compute the `filteredRacks` ref as soon as the content of `rackDetails` array changes.